framework/contracts/account/src/queries.rs

```diff
@@ -73,8 +76,11 @@ pub fn handle_module_info_query(
 
     let ids_and_addr = res?;
 
-    let registry =
-        RegistryContract::new(deps.api, env).map_err(|e| StdError::generic_err(e.to_string()))?;
+    let contract_info = deps
```

Comment:
why so much code repeats !!

Reply:
It's not like we can avoid it that much in our native contracts, we need to get contract info and code info. For modules it's abstracted away!

Reply:
We could have a method that does `Registry::new_from_self_code_id` os something no ?

Reply:
> We could have a method that does `Registry::new_from_self_code_id` os something no ?

Then we can't re-use code-id

Reply:
ok got it